src/org/openstreetmap/josm/gui/tagging/presets/TaggingPreset.java

```diff
@@ -605,6 +601,25 @@ public int showDialog(Collection<OsmPrimitive> sel, boolean showNewRelation) {
 
         int answer = 1;
         boolean canCreateRelation = types == null || types.contains(TaggingPresetType.RELATION);
+        if(regions!=null && !sel.isEmpty()){
+            boolean flag=true;
+            for(OsmPrimitive osm : sel){
+                for(String region : regions){
+
+                    //TODO iterate through the nodes of way and relation
+                    /*if(!isIso3166Code(region, osm.getCoor())){
+                        flag=false;
+                    }*/
+                }
+            }
+            if(flag){
+                new Notification(
+                        tr("The preset <i>{0}</i> cannot be applied to this region!", getLocaleName()))
+                        .setIcon(JOptionPane.WARNING_MESSAGE)
+                        .show();
+                return DIALOG_ANSWER_CANCEL;
```

Comment:
As was discussed on the call, we _want_ to allow users to apply presets that may not be valid outside of a specific region, but we do want to notify them of that. Returning `DIALOG_ANSWER_CANCEL` probably isn't the best decision for this.